### ISSUE TITLE
Database compatible with Nixnote or Evernote Wine?

### ISSUE THREAD

**Issue by bilalqayum** [nvbn/everpad#302]
Hi, I love Everpad's integration with Unity, the Dash and appindicator are genuinely awesome. I've had serious issues with sync though, like many people here, and although roughly two-thirds of my notes are synced, any new notes created in Everpad do not reach the server and therefore are not available on my mobile, web or official clients. Instead I use the Evernote official client in CrossOver for editing and Everpad for quick reference and search.

This set-up would work fine for me if Everpad sync'd fully. I'm not going to add to the issues already opened here about the sync, but rather ask if it's possible to point Everpad in the direction of another database, such as the official client's local data or NixNote? 

Both databases are SQLite3 although the size of the official client's database suggests that they've embedded images and pdf's unlike Everpad which stores them in an external directory by the look of things. If there is a clean way to hook into that database, would this not completely solve all the sync problems? I'm not a programmer, so please forgive me if this technically silly.

**Comment by duizendnegen** [comment]
No, the databases are unfortunately not compatible (or it is not as easy as that)
I am looking in to syncing as we speak, because it is annoying me to a great deal as well. I hope I can give you some good news soon.

**Comment by duizendnegen** [comment]
If I understand correctly notes never sync fully in your client, right?
Can you run, in a terminal
`killall everpad everpad-provider`
and then in two seperate terminals
`everpad-provider`
and
`everpad`
then try to sync and report on output shown in the terminal? I'm really trying to get a hold on what the sync problems are caused by. Maybe @nvbn has a more specific notion?

**Comment by bilalqayum** [comment]
Sure. In general I've found that since I reinstalled Everpad, new notes created in the webclient and the official client running under Wine sync okay, but notes created in Everpad do not sync back to the server, and there are about 20 or 30 notes that have never sync'd to Everpad.

Killing everpad and everpad-provider then sync-ing results in this below in the terminal running everpad

bilal@o2wireless:~$ everpad
Got bus address:  "unix:abstract=/tmp/dbus-T59oC2jRcE,guid=25dd953b82876bd4dd27f0215158a12c" 
Connected to accessibility bus at:  "unix:abstract=/tmp/dbus-T59oC2jRcE,guid=25dd953b82876bd4dd27f0215158a12c" 
Registered DEC:  true 
Registered event listener change listener:  true 

And in everpad-provider --verbose results in the below

bilal@o2wireless:~$ everpad-provider --verbose
Notebook bilalqayum's notebook remote

{list of most of my notebooks and notes}

Note Everpad Sync Test local

And then just stops and doesn't display anything else.

I've just doubled checked in the webclient. The Test note I created in Everpad actually did sync this time, although it does frequently fail to do so, but about 10 notebooks are still stubbornly refusing to fully sync to Everpad and the 30-odd missing notes do not appear in the list in everpad-provider --verbose even if it's left running constantly.

Hope this is of some help.

**Comment by duizendnegen** [comment]
So if I understand it correctly there's a few notebooks and notes that are available online, but aren't synced to Everpad. Is there any unique feature about those 30 notes? Could you maybe share one of them with me so I can see whether I can reproduce the same thing?
And about the Notebooks, are they all Stack Notebooks (i.e. under a stack) or is there anything else unique about them?

Thanks a lot for your help -- like I said, I'm really trying to get this thing down, because smooth syncing is the best thing about Evernote imo, and what I have to do now and then (remove the database, re-sync everything) is just impractical.

**Comment by bilalqayum** [comment]
There isn't anything special about the notes that refuse to sync as far as I can tell, I rarely use any of the 'special' feature of Evernote, just saved articles using the Clearly add-on for Firefox and reams of plain text-notes. All of my notebooks are in stacks but at least one folder in each of my stacks has a note or few missing.

Here's a saved wiki, done using Clearly on Firefox. I've got a handful of these saved for reference and this is the only one which stubbornly refuses to sync to Everpad.

http://www.evernote.com/shard/s30/sh/522d41ba-85a7-4ac2-925e-72adc0d3a82f/f316a956bd2268fae19b22b1bbab92b4

**Comment by duizendnegen** [comment]
Hmm, I have clipped the same article with Clearly & attempted to Sync, but it does so without any problem. I have also used "Save to Evernote" to copy your note and move it to my own note store, but sync works without any problem again.
So unfortunately this does not give me a handle on what is causing the sync issues yet. Maybe a note that is processed "before" the Magic SysRq key note is causing the problems, but it's hard to find out which one that'd be since the note doesn't even appear in the everpad-provider sync trace. Hm.

**Comment by bilalqayum** [comment]
If it's any help, I just check my Everpad main window again, and I did notice one thing, the unsync'd notes are completely missing, as in not searchable in the Dash and not in any notebook or stack, but there are a lot I've noticed that are in my default notebook when they should have been moved into others.

I'll have to check thoroughly later in this evening, but I've got a feeling that maybe all the missing notes are one that were clipped from the web, those notes go into my default notebook and all of them are then moved into other notebooks, the default one remains empty unless something was recently clipped. I usually just search for notes through the Dash so I've not noticed how many notes have been piling up in my default notebook before.

It's maybe a thin lead, but I'm pretty sure that all the notes that I've created myself and sorted into notebooks when they were made are present and accounted for, if so then maybe it'll narrow down exactly which types of notes have trouble syncing.
